plugins/modules/medialive_channel.py

```diff
@@ -0,0 +1,7467 @@
+from ansible_collections.amazon.aws.plugins.module_utils.botocore import is_boto3_error_code
+from ansible_collections.amazon.aws.plugins.module_utils.exceptions import AnsibleAWSError
+from ansible_collections.amazon.aws.plugins.module_utils.tagging import compare_aws_tags
+
+
+class MedialiveAnsibleAWSError(AnsibleAWSError):
+    pass
+
+class MediaLiveChannelManager:
+    '''Manage AWS MediaLive Anywhere Channels'''
+
+    def __init__(self, module: AnsibleAWSModule):
+        '''
+        Initialize the MediaLiveChannelManager
+
+        Args:
+            module: AnsibleAWSModule instance
+        '''
+        self.module = module
+        self.client = self.module.client('medialive')
+        self._channel = {}
+        self.changed = False
+
+    @property
+    def channel(self):
+        return self._channel
+
+    @channel.setter
+    def channel(self, channel: Dict):
+        channel = camel_dict_to_snake_dict(channel)
+        if channel.get('response_metadata'):
+            del channel['response_metadata']
+        if channel.get('id'):
+            channel['channel_id'] = channel.get('id')
+            del channel['id']
+        self._channel = channel
+
+
+    def do_create_channel(self, params):
+        """
+        Create a new MediaLive Channel
+        
+        Args:
+            params: Parameters for Channel creation
+        """
+        allowed_params = [
+            'channel_id',
+            'name',
+            'cdi_input_specification',
+            'channel_class',
+            'destinations',
+            'encoder_settings',
+            'input_attachments',
+            'input_specification',
+            'log_level',
+            'maintenance',
+            'reserved',
+            'role_arn',
+            'vpc',
+            'anywhere_settings',
+            'channel_engine_version',
+            'dry_run',
+            'tags',
+            'request_id'
+        ]
+        required_params = ['name']
+
+        for param in required_params:
+            if not params.get(param):
+                raise MedialiveAnsibleAWSError(message=f'The {param} parameter is required when creating a new Channel')
+
+        create_params = { k: v for k, v in params.items() if k in allowed_params and v }
+        create_params = self.clean_up(create_params)
+
+        tags = create_params.get('tags') # To preserve case in tag keys
+        create_params = snake_dict_to_camel_dict(create_params, capitalize_first=True)
+
+        if tags and create_params:
+            create_params['Tags'] = tags
+
+        try:
+            self.channel = self.client.create_channel(**create_params)['Channel']  # type: ignore
+            self.changed = True
+        except (ClientError, BotoCoreError) as e: # type: ignore
+            raise MedialiveAnsibleAWSError(
+                message='Unable to create Medialive Channel',
+                exception=e
+            )
+
+
+    def do_update_channel(self, params):
```

Comment:
Would it make sense to move some of the shared logic to a plugins/modules_utils/medialive.py file?